extension/data.ts

```diff
@@ -86,6 +86,27 @@ interface DeinflectionRuleGroup {
   rules: DeinflectionRule[];
 }
 
+const WORD_DICT_ENTRY_REGEX = /^(.+?)\s+(?:\[(.*?)\])?\s*\/(.+)\//;
+
+type WordDictEntry = {
+  word: string;
+  pronunciation: string | null;
+  definitions: string[];
+};
+
+export const parseWordDictEntry = (line: string): WordDictEntry | null => {
```

Comment:
In this case, we're exporting this function soley in order to test it but in the end increases it visibility for any other use case as well.

In order to avoid API creep, a good best practice is to have all methods/variables as non-accessible as possible for the needs of the software.

In this case, is there any reason this method can't be private inside of RcxDict? Isn't all of this logic already tested via the public APIs of `makeText`?

If we did want a generic entry parsing function, I'd recommend putting it in its own module so that it can be treated and maintained as an independent utility.

-- Whoops! I just remembered that you did say you planned to use it in update-db.ts as well so in that case I would go with the last recommendation and extract it to it's own module. This will keep files small and make them very easy to reason about; often called the Single Responsibilty Principle.

Reply:
I put this in its own file so it can be used in update-db.ts later.